### PR TITLE
Change detector integrate

### PR DESCRIPTION
**Pull request by kegluneq**

### CODE REVIEW

File: modules/angular2/src/change_detection/change_detection.ts [hunk 1]
```diff
@@ -68,17 +68,20 @@ export var defaultPipes = {
 
 export var preGeneratedProtoDetectors = {};
 
+export const PROTO_CHANGE_DETECTOR_KEY = 'ProtoChangeDetectors';
```
Comment (vsavkin) [inline modules/angular2/src/change_detection/change_detection.ts]: You should use something like OpaqueToken here.

File: modules/angular2/src/core/application.ts [hunk 2]
```diff
@@ -118,7 +118,7 @@ function _injectorBindings(appComponentType): List<Type | Binding | List<any>> {
     CompilerCache,
     TemplateResolver,
     bind(PipeRegistry).toValue(defaultPipeRegistry),
-    bind(ChangeDetection).toClass(DynamicChangeDetection),
+    bind(ChangeDetection).toClass(PreGeneratedChangeDetection),
```
Comment (vsavkin) [inline modules/angular2/src/core/application.ts]: To enable this we need to do the following two things:

* Ports all tests to make sure that the new detector actually works the same way.
* Enable JIT detection for javascript. We should do that by conditionally setting the detector: JIT for JS, PreGenerated for Dart.

Comment (vsavkin) [inline modules/angular2/src/core/application.ts]: To enable this we need to do the following two things:

* Ports all tests to make sure that the new detector actually works the same way.
* Enable JIT detection for javascript. We should do that by conditionally setting the detector: JIT for JS, PreGenerated for Dart.

**Comment by naomiblack** [comment]
Also looks like tests are failing on clang-format. Make sure you've updated to 1.0.15

**Comment by vsavkin** [comment]
@kegluneq could you update this PR?

**Comment by kegluneq** [comment]
Yep, this is my next focus. I just pushed an update incorporating all the recent changes to pre-generated change detectors -- now I want to work on properly enabling the pregen vs. JIT change detectors depending on when they're available.

**Pull request merged**